driver/controller.go

```diff
@@ -81,6 +85,15 @@ func (s controllerService) CreateVolume(ctx context.Context, req *csi.CreateVolu
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// check if the create volume request has a data source
+	if source != nil {
+		// get the parent volumeID/snapshotID if exists
+		parentVol, parentID, err = s.validateContentSource(ctx, req)
```

Comment:
- do we need `parentID` as well, IMHO we can just use `parentVol.Status.VolumeID` for both the scenarios
- correct me I'm missing something here

Reply:
We are now just passing parentID to the latter functions, and using parentVol to fetch some information from the parent volume.